### DISCUSSION TITLE
HMR hangs for a couple of seconds when use BundleAnalyzerPlugin in development mode

### DISCUSSION THREAD

**Discussion by constgen** [webpack-contrib/webpack-bundle-analyzer#586] [Ideas]
#### Issue description

When BundleAnalyzerPlugin  is not used in Webpack config, HMR reloads React components almost immediately. But just adding BundleAnalyzerPlugin  to the config adds several extra seconds to the webpage to react on Hot Module Replacement. Can we some how make operations of this plugin async in case of HMR?

**Comment by valscion** [comment]
Hi, thanks for opening an issue ☺️ 

As for your question, I unfortunately have no idea what "making operations of this plugin async in case of HMR" could look like. Would you happen to be able to find any example of a webpack plugin that would allow such behavior?

**Comment by constgen** [comment]
I have nave faced anything like this. But may be on initial build it should work as it is now, but on the next incremental re-build this plugin can call `callback()` immediately and then do its job without blocking the thread. We need it in the watch mode just to livereload the webserver with the bundle analyzes chart. But it not necessarily needs to block the main thread where the application is built.

**Comment by valscion** [comment]
I'm not sure if that's possible to do without throwing away the "bundle analyzer UI updates automatically after compilation" feature here.

**Comment by constgen** [comment]
I would like to investigate it. Can you give me some hints? Is this true that "bundle analyzer UI updates automatically after compilation" feature works on HMR client and is triggered by the Webpack itself but not this plugin?

**Comment by th0r** [comment]
@constgen [this callback](https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/d09366690257ca69b2b0639cf20f91c0a46f0c0c/src/BundleAnalyzerPlugin.js#L37) is called on every bundle recompilation. So, for `analyzerMode: 'server'` it calls [startAnalyzerServer](https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/d09366690257ca69b2b0639cf20f91c0a46f0c0c/src/BundleAnalyzerPlugin.js#L98) method which calls [updateChartData](https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/d09366690257ca69b2b0639cf20f91c0a46f0c0c/src/BundleAnalyzerPlugin.js#L100) method to push changes to the client via websocket.
